python/cuml/cuml/manifold/umap.pyx

```diff
@@ -273,10 +273,16 @@ class UMAP(UniversalBase,
         'nn_descent']. 'auto' chooses to run with brute force knn if number of data rows is
         smaller than or equal to 50K. Otherwise, runs with nn descent.
     build_kwds: dict (optional, default=None)
-        Build algorithm argument {'nnd_graph_degree': 64, 'nnd_intermediate_graph_degree': 128,
-        'nnd_max_iterations': 20, 'nnd_termination_threshold': 0.0001, 'nnd_return_distances': True,
-        'nnd_n_clusters': 1}
-        Note that nnd_n_clusters > 1 will result in batch-building with NN Descent.
+        Build algorithm argument. Default values are: {'n_clusters': 1, 'n_nearest_clusters':2, 'nn_descent': {'graph_degree': n_neigbors, 'max_iterations': 20}}.
+       "n_clusters": int (default=1). Number of clusters to split the data into when building the knn graph. Increasing this will use less device memory at the cost of accuracy. When using n_clusters > 1, put data on host (refer to data_on_host argument for fit_transform). The default value (n_clusters=1) will place the entire data on device memory.
```

Comment:
> put data on host

Is this a requirement?

Reply:
Yes. There is no point of using batching (n_clusters > 1) if we end up putting the entire data on GPU memory. The current all-neighbors wrapper raises an error and informs the user what to do if the user attempts to use n_clusters>1 AND data on GPU memory.

Reply:
Can you just spell it out clearly in the docs here that it is a hard requirement?